superset/assets/src/visualizations/FilterBox/FilterBox.jsx

```diff
@@ -178,8 +178,7 @@ class FilterBox extends React.Component {
     // Add created options to filtersChoices, even though it doesn't exist,
     // or these options will exist in query sql but invisible to end user.
     Object.keys(selectedValues)
-      .filter(key => !selectedValues.hasOwnProperty(key)
-        || !(key in filtersChoices))
+      .filter(key => selectedValues.hasOwnProperty(key))
```

Comment:
Add `&& (key in filtersChoices)`

Thanks for fixing this. This was due to my refactor (#5789) that reversed the logic so the correct logic should be negation of 
```js
!selectedValues.hasOwnProperty(key) || !(key in filtersChoices)
```

which is

```js
selectedValues.hasOwnProperty(key) && (key in filtersChoices)
```